design-proposals/project-deploymnet.md

```diff
@@ -0,0 +1,176 @@
+
+### Create Fabric certificate
+Create the fabric certificate, which is the root CA certificate for all the K8s clusters:
+
+    clusterlink create fabric --name <fabric_name>
+
+The root ca certificates will be saved in the directory ```~/.config/clustrlink/<fabric_name>```.
+
+### Install ClusterLink on K8s Cluster
+The easiest way to install ClusterLink on the K8s cluster is by using the command:
+
+    clusterlink install --peer <name>
+
+This command installs the ClusterLink operator and creates the ClusterLink deployment.  
+Alternatively you can do it in steps:
+First, create and deploy the peer certificate:
+
+    clusterlink create peer --name <name>
+    kubectl create -f ```~/.config/clustrlink/<fabric_name>/<peer_nam>/secret.yaml```
+
+Second, create the ClustreLink operator:
+
+    kubectl create -f config/setup/
+
+Then, create clusterlink deployment:
+```
+kubectl apply -f - <<EOF
+apiVersion: clusterlink.net/v1alpha1
+kind: ClusterLink
+metadata:
+  namespace: clusterlink-ns
+  name: peer1
+spec:
+  dataplaneType: "envoy"
+  dataplaneReplicates: 1
+EOF
+```
+The operator will create:
+* ClusterLink controlplane deployment, including controlplane-pod, controlplane-service, and RBAC roles.  
+* ClusterLink dataplane deployment, including dataplane-pod and dataplane-service.
+* ClusterLink external access point using load-balancer/node-port/gateway-API service.
+  
+Overall, the ClusterLink deployment stages are:
+
+<img src="deployment.png" width="800" height="400" alt="ClusterLink Deployment Stages"/>    
+
+
+
+## Goals
+This design document should:
+
+- Define the steps for deploying a ClusterLink gateway to a K8s cluster.
+- Define the task functions of the ClusterLink operator.
+- Define the CRD (Custom Resource Definition) for the ClusterLink operator.
+
+## Non-Goals
+The following aspects are explicitly excluded and are out of scope in the current design:
+
+- This document focuses only on the deployment of ClusterLink to the K8s cluster. 
+The deployment to another environment as VMs is out of scope. 
+- The security aspects and user permissions concerning the creation of certificates, specifically for fabric or peer.
+## Proposal
+The ClusterLink deployment will be done in the following steps, (as described in more detail in the User section):
+
+1. Create a Fabric certificate if it does not exist.
+2. Create the ClusterLink operator.
+3. Deploy the ClusterLink CRD to the operator, which will create peer certificates, ClusterLink deployments (control plane and data plane), services, and external access point.
+
+### ClusterLink CRD
+The ClusterLink CRD includes the following fields:
+- Api version: clusterlink.net/v1alpha1
+- Kind: clusterlink
+
+- **Spec Section:**
+  
+    | Field name | Description | Default value |
+    | ---- | ----- | ------ |
+    |dataplaneType| Types of dataplane, supported values "go" or "envoy"|"envoy"|
+    |dataplaneReplicates| Number of dataplane replicas|1|
```

Comment:
dataplaneReplicas? 
what do you think about splitting the spec into shared values for both data and control and then having Spec.DataPlane and Spec.ControlPlane objects for the specific deployment (e.g., we would then have `Spec.DataPlane.replicas` but `image` might be shared)